pkgs/operator/state.go

```diff
@@ -220,11 +309,139 @@ func (s *Switch) InitInstance(reqID [24]byte, initMsg *wire.Transport, initiator
 	s.Instances[reqID] = inst
 	s.InstanceInitTime[reqID] = time.Now()
 	s.Mtx.Unlock()
-
 	return resp, nil
+}
 
+func (s *Switch) InitInstanceReshare(reqID [24]byte, reshareMsg *wire.Transport, initiatorSignature []byte) ([]byte, error) {
+	logger := s.Logger.With(zap.String("reqid", hex.EncodeToString(reqID[:])))
+	logger.Info("🚀 Initializing DKG instance")
+	reshare := &wire.Reshare{}
+	if err := reshare.UnmarshalSSZ(reshareMsg.Data); err != nil {
+		return nil, err
+	}
+	// Check that incoming init message signature is valid
+	initiatorPubKey, err := crypto.ParseRSAPubkey(reshare.InitiatorPublicKey)
+	if err != nil {
+		return nil, err
+	}
+	marshalledWireMsg, err := reshareMsg.MarshalSSZ()
+	if err != nil {
+		return nil, err
+	}
+	err = crypto.VerifyRSA(initiatorPubKey, marshalledWireMsg, initiatorSignature)
+	if err != nil {
+		return nil, fmt.Errorf("init message: initiator signature isn't valid: %s", err.Error())
+	}
+	initiatorID := sha256.Sum256(initiatorPubKey.N.Bytes())
+	s.Logger.Info("✅ reshare message signature is successfully verified", zap.String("from initiator", fmt.Sprintf("%x", initiatorID[:])))
+
+	s.Logger.Info("Starting resharing protocol")
+	// try to get old share local owner first
+	var shareFromDB basedb.Obj
+	secret := &kyber_dkg.DistKeyShare{}
+	shareFromDB, ok, err := s.DB.Get([]byte("secret"), reshare.OldID[:])
+	if err != nil {
+		return nil, err
+	}
+	if ok {
```

Comment:
we aren't going to reshare if there's no secret in db right? lets do 
```golang
if !ok {
 fmt.Errorf("can't find secret share in db for id %v", reshare.OldID[:]) 
}
```

Reply:
Not exactly, new operators may not have an old share in DB so we are creating an instance without a share